BeeJet.Bot/Commands/Handlers/Steam/SyncSteamCommandHandler.cs

```diff
@@ -8,27 +10,44 @@ namespace BeeJet.Bot.Commands.Handlers.Steam
     public class SyncSteamCommandHandler : CommandSource
     {
         private SteamAPIService _steamAPI;
+        private readonly ISteamIdDiscordUserDb _steamUserDb;
+        private readonly BeeJetBotOptions _beeJetOptions;
 
-        public SyncSteamCommandHandler(Services.SteamAPIService steamAPI)
+        public SyncSteamCommandHandler(Services.SteamAPIService steamAPI, ISteamIdDiscordUserDb steamUserDb, BeeJetBotOptions options)
         {
             _steamAPI = steamAPI;
+            _steamUserDb = steamUserDb;
+            _beeJetOptions = options;
         }
 
         [BeeJetBotSlashCommand("sync-steam", "Sync steam library with channels", nameof(RegisterOptions))]
         public async Task SlashCommandExecuted()
         {
-            if (!ulong.TryParse((string)Context.SlashCommandInteraction.Data.Options.First().Value, out ulong steamId))
+            if (Context.SlashCommandInteraction.Data.Options.Count > 0 && !ulong.TryParse((string)Context.SlashCommandInteraction.Data.Options.First().Value, out ulong steamId))
             {
                 await Context.SlashCommandInteraction.RespondAsync("Not a valid steamid", ephemeral: true);
                 return;
             }
+            else
+            {
+                var steamIdFromDb = _steamUserDb.GetSteamId(Context.User.Id.ToString());
+                if (string.IsNullOrWhiteSpace(steamIdFromDb) || !ulong.TryParse(steamIdFromDb, out steamId))
+                {
+                    EmbedBuilder embed = new EmbedBuilder();
```

Comment:
Ik zou de binnenkant van deze if afsplitsen naar losse method voor de leesbaarheid (plus die kunnen we misschien nog wel vaker inzetten)